packages/core/utils/lib/validators.js

```diff
@@ -56,27 +57,11 @@ class StrapiIDSchema extends MixedSchemaType {
 
 yup.strapiID = () => new StrapiIDSchema();
 
-/**
- * Returns a formatted error for http responses
- * @param {Object} validationError - a Yup ValidationError
- */
-const formatYupErrors = validationError => {
-  if (!validationError.inner) {
-    throw new Error('invalid.input');
-  }
-
-  if (validationError.inner.length === 0) {
-    if (validationError.path === undefined) return validationError.errors;
-    return { [validationError.path]: validationError.errors };
-  }
-
-  return validationError.inner.reduce((acc, err) => {
-    acc[err.path] = err.errors;
-    return acc;
-  }, {});
+const handleYupError = error => {
+  throw new YupValidationError(error);
 };
 
 module.exports = {
   yup,
-  formatYupErrors,
+  handleYupError,
```

Comment:
I think we should go one step further and export a validation function directly to wrap all this catching logic

```js
const validateYupSchema = (schema, options = { strict: true, abortEarly: false ) => { 
  return schema.validate(options).catch(err => handleYupError)
}

const validateYupSchemaSync = (schema, options = { strict: true, abortEarly: false ) => { 
  try {
    return schema.validateSync(options)
  } catch (error) {
    handleYupError(error)
  }
}
```

Reply:
Yes I was thinking the same

Reply:
we can do that a bit later after merging this first pass